base/data/Campaign.js

```diff
@@ -86,5 +88,169 @@ Campaign.hideCharities = campaign => {
 	return hideCharitiesArr;
 };
 
+/**
+ * Get the list of ad IDs that this campaign will hide
+ * Optional: also merge with lists from other campaigns
+ * @param {Campaign} topCampaign 
+ * @param {?Campaign[]} campaigns 
+ * @returns {String[]} hideAdverts
+ */
+Campaign.hideAdverts = (topCampaign, campaigns) => {
+    
+    // Merge all hide advert lists together from all campaigns
+    let allHideAds = topCampaign.hideAdverts ? keysetObjToArray(topCampaign.hideAdverts) : [];
+    if (campaigns) campaigns.forEach(campaign => allHideAds.push(... campaign.hideAdverts ? keysetObjToArray(campaign.hideAdverts) : []));
+    // Copy array
+    const mergedHideAds = allHideAds.slice();
```

Comment:
FYI - a defensive copy of the array is harmless, but not needed. The array is freshly made within this function, and any edits to it won't affect the topCampaign object.